ipv8/test/base.py

```diff
@@ -19,12 +18,6 @@
 from ..peer import Peer
 from ..types import Endpoint, AnyPayloadType, Community, AnyPayload
 
-try:
-    get_event_loop().set_debug(True)
-except RuntimeError:
-    logging.warning("Failed to set debug mode on the main event loop! "
-                    "You may be missing out on asyncio output!")
-
```

Comment:
Could you keep this functionality in the code? Setting any existing loop to debug mode was a feature introduced in #1003 and apparently found several suppressed bugs, described in https://github.com/Tribler/tribler/issues/6087

Reply:
On second thought, I agree with your deletion after all: we'll get a lot of debug info spewed into the terminal that cannot be redirected (because the test case instances have not been "setUpClass"-ed yet). If a programmer already has a running event loop, it's their responsibility to set debug mode outside of testing. If any tests miraculously fail, programmers can always manually enable debug on a running loop.

I'll leave it to you to "resolve conversation" if you agree with leaving your change as-is.

Reply:
OK, then I'll remove this call to `set_debug`. I know you're already convinced, but since my reasons for removing this seem to be different to yours, let me clarify my thinking.

In Python 3.8 - 3.10 `IsolatedAsyncioTestCase` calls [`loop.set_debug(True)`](https://github.com/python/cpython/blob/3.8/Lib/unittest/async_case.py#L115). Python 3.11+ it uses [`Runner(debug=True)`](https://github.com/python/cpython/blob/3.11/Lib/unittest/async_case.py#L121). So, when using Python 3.8+ the debug mode is already turned on. In `asynctest` (needed for Python 3.7 support), the event loop doesn't run in debug mode by default. However, even here, it still doesn't seem needed since [debug mode is turned on again](https://github.com/Tribler/py-ipv8/blob/master/ipv8/test/base.py#L224) in `setUp`.

Additionally, `get_event_loop` gets the current loop, while the unit test stuff (both `asyncio` and `asynctest`) always seems to create a brand new loop without copying the debug value from the previous loop. So, if we want to turn debug mode on, this wouldn't be the place to do it. This block of code effectively does nothing as far as I can tell (other than perhaps debugging the unittest module itself).

I haven't read through the entire issue you mentioned, but I'm guessing that you were using `asynctest` and that adding `loop.set_debug(True)` to `setUp` allowed you to find these hidden bugs, not this block of code.

Reply:
Mostly agreed. However, this code did do _something_ in the past. Apparently, the current version is broken though 😕, which is even more reason to just remove this. Fixing it using the following patch shows what is lost: 

<details> <summary>[click me to expand] Patch file</summary>

```patch
diff --git a/ipv8/test/base.py b/ipv8/test/base.py
index 653f09d..e8f42c0 100644
--- a/ipv8/test/base.py
+++ b/ipv8/test/base.py
@@ -7,7 +7,7 @@ import threading
 import time
 import unittest
 import uuid
-from asyncio import all_tasks, ensure_future, get_event_loop, iscoroutine, sleep
+from asyncio import all_tasks, ensure_future, get_event_loop, get_running_loop, iscoroutine, sleep
 from contextlib import contextmanager
 from functools import partial
 from typing import List, Optional
@@ -20,7 +20,7 @@ from ..peer import Peer
 from ..types import Endpoint, AnyPayloadType, Community, AnyPayload
 
 try:
-    get_event_loop().set_debug(True)
+    get_running_loop().set_debug(True)
 except RuntimeError:
     logging.warning("Failed to set debug mode on the main event loop! "
                     "You may be missing out on asyncio output!")
```

</details>

<details> <summary>[click me to expand] New output (otherwise this would just be empty between "run_all_tests.py" and "Summary:".</summary>

```
$ python3 run_all_tests.py -q -a
WARNING:root:Failed to set debug mode on the main event loop! You may be missing out on asyncio output!
Launching in 8 processes ... awaiting results ... WARNING:IdentityCommunity:Cannot anonymize community traffic without TunnelEndpoint
WARNING:IdentityCommunity:Cannot anonymize community traffic without TunnelEndpoint
WARNING:IdentityCommunity:Cannot anonymize community traffic without TunnelEndpoint
WARNING:IdentityCommunity:Cannot anonymize community traffic without TunnelEndpoint
ERROR:root:Snapshot loading got stuck! Aborting snapshot load.
ERROR:root:Snapshot loading got stuck! Aborting snapshot load.
ERROR:HiddenTunnelCommunity:No circuit for contacting the introduction point
ERROR:root:Exception occurred while guessing interface for ('2001:db8:::ff00:42:8329', 5)
Traceback (most recent call last):
  File "/home/quinten/Documents/py-ipv8/ipv8/messaging/interfaces/dispatcher/endpoint.py", line 47, in guess_interface
    if isinstance(ipaddress.ip_address(socket_address[0]), ipaddress.IPv4Address):
  File "/usr/lib/python3.10/ipaddress.py", line 54, in ip_address
    raise ValueError(f'{address!r} does not appear to be an IPv4 or IPv6 address')
ValueError: '2001:db8:::ff00:42:8329' does not appear to be an IPv4 or IPv6 address
WARNING:IdentityCommunity:Cannot anonymize community traffic without TunnelEndpoint
WARNING:IdentityCommunity:Cannot anonymize community traffic without TunnelEndpoint
WARNING:IdentityCommunity:Cannot anonymize community traffic without TunnelEndpoint
WARNING:IdentityCommunity:Cannot anonymize community traffic without TunnelEndpoint
WARNING:asyncio:Executing <Task finished name='AttestationCommunity:on_packet 1' coro=<AttestationCommunity.on_request_attestation() done, defined at /home/quinten/Documents/py-ipv8/ipv8/attestation/wallet/community.py:169> result=None created at /usr/lib/python3.10/asyncio/tasks.py:636> took 0.169 seconds
WARNING:TunnelCommunity:Received unexpected created for circuit 3651316213
WARNING:IdentityCommunity:Cannot anonymize community traffic without TunnelEndpoint
WARNING:IdentityCommunity:Cannot anonymize community traffic without TunnelEndpoint
WARNING:IdentityCommunity:Received unsolicited disclosure from Peer<53.53.176.6:32450, r9OlU7SmabQK/D3ltA9FWRMqKtQ=>, dropping!
WARNING:IdentityCommunity:Cannot anonymize community traffic without TunnelEndpoint
WARNING:IdentityCommunity:Cannot anonymize community traffic without TunnelEndpoint
WARNING:IdentityCommunity:Cannot anonymize community traffic without TunnelEndpoint
WARNING:IdentityCommunity:Cannot anonymize community traffic without TunnelEndpoint
WARNING:IdentityCommunity:Received unsolicited disclosure from Peer<165.220.168.213:17145, cgNa/S4h/YQ+Je3V5w7O5+yG9GA=>, dropping!
WARNING:IdentityCommunity:Cannot anonymize community traffic without TunnelEndpoint
WARNING:IdentityCommunity:Cannot anonymize community traffic without TunnelEndpoint
WARNING:IdentityCommunity:Cannot anonymize community traffic without TunnelEndpoint
WARNING:IdentityCommunity:Cannot anonymize community traffic without TunnelEndpoint
WARNING:TunnelCommunity:Dropping cell (too many relay_early cells)
WARNING:TaskManager:Not adding task <function TestTaskManager.test_shutdown.<locals>.<lambda> at 0x7fdba08c5630> due to shutdown!
WARNING:IdentityCommunity:Cannot anonymize community traffic without TunnelEndpoint
ERROR:TaskManager:Task resulted in error: division by zero
Traceback (most recent call last):
  File "/home/quinten/Documents/py-ipv8/ipv8/taskmanager.py", line 131, in done_cb
    future.result()
  File "/home/quinten/Documents/py-ipv8/ipv8/util.py", line 35, in call_async
    return func(*args, **kwargs)
  File "/home/quinten/Documents/py-ipv8/ipv8/test/test_taskmanager.py", line 133, in <lambda>
    await self.tm.register_task('test', lambda: 1 / 0, ignore=(ZeroDivisionError,))
ZeroDivisionError: division by zero

ERROR:RequestCache:add with duplicate identifier "mock:30020"
WARNING:RequestCache:Dropping <MockRegisteredCache mock-48699> due to shutdown!
WARNING:RequestCache:Dropping <MockCache mock-11666> due to shutdown!
WARNING:asyncio:Executing <Task finished name='AttestationCommunity:on_packet 1' coro=<AttestationCommunity.on_request_attestation() done, defined at /home/quinten/Documents/py-ipv8/ipv8/attestation/wallet/community.py:169> result=None created at /usr/lib/python3.10/asyncio/tasks.py:636> took 0.700 seconds
WARNING:IdentityCommunity:Cannot anonymize community traffic without TunnelEndpoint
WARNING:asyncio:Executing <Task pending name='Task-70' coro=<IsolatedAsyncioTestCase._asyncioLoopRunner() running at /usr/lib/python3.10/unittest/async_case.py:95> wait_for=<Future pending cb=[Task.task_wakeup()] created at /usr/lib/python3.10/asyncio/base_events.py:429> created at /usr/lib/python3.10/unittest/async_case.py:117> took 0.496 seconds
WARNING:TunnelCommunity:Cannot remove unknown circuit 2691120512
WARNING:asyncio:Executing <Task finished name='AttestationCommunity:on_packet 1' coro=<AttestationCommunity.on_request_attestation() done, defined at /home/quinten/Documents/py-ipv8/ipv8/attestation/wallet/community.py:169> result=None created at /usr/lib/python3.10/asyncio/tasks.py:636> took 0.101 seconds
WARNING:asyncio:Executing <Task finished name='Task-82' coro=<RequestHandler._handle_request() done, defined at /home/quinten/.local/lib/python3.10/site-packages/aiohttp/web_protocol.py:423> result=(<Response OK eof>, False) created at /home/quinten/.local/lib/python3.10/site-packages/aiohttp/web_protocol.py:508> took 0.562 seconds
WARNING:asyncio:Executing <Task finished name='AttestationCommunity:on_packet 1' coro=<AttestationCommunity.on_request_attestation() done, defined at /home/quinten/Documents/py-ipv8/ipv8/attestation/wallet/community.py:169> result=None created at /usr/lib/python3.10/asyncio/tasks.py:636> took 0.100 seconds
WARNING:IdentityCommunity:Cannot anonymize community traffic without TunnelEndpoint
WARNING:IdentityCommunity:Cannot anonymize community traffic without TunnelEndpoint
WARNING:asyncio:Executing <Task pending name='Task-94' coro=<IsolatedAsyncioTestCase._asyncioLoopRunner() running at /usr/lib/python3.10/unittest/async_case.py:95> wait_for=<Future pending cb=[Task.task_wakeup()] created at /usr/lib/python3.10/asyncio/base_events.py:429> created at /usr/lib/python3.10/unittest/async_case.py:117> took 0.180 seconds
WARNING:asyncio:Executing <Handle Endpoint.notify_listeners((UDPv4Address(...', port=55696), b'\x00\x02\xb...01\n3\xf6\x02')) created at /home/quinten/Documents/py-ipv8/ipv8/test/mocking/endpoint.py:53> took 1.021 seconds
WARNING:asyncio:Executing <Task finished name='Task-106' coro=<RequestHandler._handle_request() done, defined at /home/quinten/.local/lib/python3.10/site-packages/aiohttp/web_protocol.py:423> result=(<Response OK eof>, False) created at /home/quinten/.local/lib/python3.10/site-packages/aiohttp/web_protocol.py:508> took 0.434 seconds
WARNING:TunnelCommunity:We're not joining circuit with ID 4180283830
WARNING:IdentityCommunity:Cannot anonymize community traffic without TunnelEndpoint
WARNING:IdentityCommunity:Cannot anonymize community traffic without TunnelEndpoint
WARNING:HiddenTunnelCommunity:Invalid or unauthorized destroy
ERROR:HiddenTunnelCommunity:UDPv4Address(ip='105.215.111.187', port=31320) not allowed send destroy for circuit 2086821492
WARNING:IdentityCommunity:Cannot anonymize community traffic without TunnelEndpoint
WARNING:asyncio:Executing <Task pending name='Task-129' coro=<IsolatedAsyncioTestCase._asyncioLoopRunner() running at /usr/lib/python3.10/unittest/async_case.py:95> wait_for=<Future pending cb=[Task.task_wakeup()] created at /usr/lib/python3.10/asyncio/base_events.py:429> created at /usr/lib/python3.10/unittest/async_case.py:117> took 0.178 seconds
WARNING:asyncio:Executing <Handle Endpoint.notify_listeners((UDPv4Address(...', port=40800), b'\x00\x02\xb...b1\x86{\x0b\t')) created at /home/quinten/Documents/py-ipv8/ipv8/test/mocking/endpoint.py:53> took 0.125 seconds
WARNING:TunnelCommunity:We're not joining circuit with ID 4115023399
WARNING:asyncio:Executing <Task finished name='Task-141' coro=<RequestHandler._handle_request() done, defined at /home/quinten/.local/lib/python3.10/site-packages/aiohttp/web_protocol.py:423> result=(<Response OK eof>, False) created at /home/quinten/.local/lib/python3.10/site-packages/aiohttp/web_protocol.py:508> took 0.470 seconds
WARNING:IdentityCommunity:Cannot anonymize community traffic without TunnelEndpoint
WARNING:IdentityCommunity:Cannot anonymize community traffic without TunnelEndpoint
WARNING:asyncio:Executing <Task pending name='Task-149' coro=<IsolatedAsyncioTestCase._asyncioLoopRunner() running at /usr/lib/python3.10/unittest/async_case.py:95> wait_for=<Future pending cb=[Task.task_wakeup()] created at /usr/lib/python3.10/asyncio/base_events.py:429> created at /usr/lib/python3.10/unittest/async_case.py:117> took 0.162 seconds
WARNING:asyncio:Executing <Task finished name='Task-161' coro=<RequestHandler._handle_request() done, defined at /home/quinten/.local/lib/python3.10/site-packages/aiohttp/web_protocol.py:423> result=(<Response OK eof>, False) created at /home/quinten/.local/lib/python3.10/site-packages/aiohttp/web_protocol.py:508> took 0.428 seconds
WARNING:asyncio:Executing <Task pending name='Task-387' coro=<IsolatedAsyncioTestCase._asyncioLoopRunner() running at /usr/lib/python3.10/unittest/async_case.py:101> wait_for=<Future pending cb=[Task.task_wakeup()] created at /usr/lib/python3.10/asyncio/base_events.py:429> created at /usr/lib/python3.10/unittest/async_case.py:117> took 0.837 seconds
WARNING:IdentityCommunity:Cannot anonymize community traffic without TunnelEndpoint
WARNING:IdentityCommunity:Cannot anonymize community traffic without TunnelEndpoint
WARNING:asyncio:Executing <Task pending name='Task-180' coro=<IsolatedAsyncioTestCase._asyncioLoopRunner() running at /usr/lib/python3.10/unittest/async_case.py:95> wait_for=<Future pending cb=[Task.task_wakeup()] created at /usr/lib/python3.10/asyncio/base_events.py:429> created at /usr/lib/python3.10/unittest/async_case.py:117> took 0.136 seconds
WARNING:IdentityCommunity:Cannot anonymize community traffic without TunnelEndpoint
WARNING:IdentityCommunity:Cannot anonymize community traffic without TunnelEndpoint
WARNING:TunnelExitSocket:Dropping data packets, refusing to be an exit node (BT=True, IPv8=False)
WARNING:TunnelExitSocket:Dropping data packets, refusing to be an exit node (BT=True, IPv8=False)
WARNING:TunnelExitSocket:Dropping data packets, refusing to be an exit node (BT=True, IPv8=False)
WARNING:TunnelExitSocket:Dropping data packets, refusing to be an exit node (BT=False, IPv8=True)
WARNING:TunnelExitSocket:Dropping data packets, refusing to be an exit node (BT=False, IPv8=True)
WARNING:TunnelExitSocket:Dropping data packets, refusing to be an exit node (BT=True, IPv8=False)
WARNING:TunnelExitSocket:Dropping data packets, refusing to be an exit node (BT=True, IPv8=False)
WARNING:TunnelExitSocket:Dropping data packets, refusing to be an exit node (BT=True, IPv8=False)
WARNING:asyncio:Executing <Task pending name='Task-194' coro=<IsolatedAsyncioTestCase._asyncioLoopRunner() running at /usr/lib/python3.10/unittest/async_case.py:95> wait_for=<Future pending cb=[Task.task_wakeup()] created at /usr/lib/python3.10/asyncio/base_events.py:429> created at /usr/lib/python3.10/unittest/async_case.py:117> took 0.194 seconds
WARNING:asyncio:Executing <Task finished name='AttestationCommunity:on_packet 1' coro=<AttestationCommunity.on_verify_attestation_request() done, defined at /home/quinten/Documents/py-ipv8/ipv8/attestation/wallet/community.py:235> result=None created at /usr/lib/python3.10/asyncio/tasks.py:636> took 0.843 seconds
WARNING:asyncio:Executing <Task finished name='Task-206' coro=<RequestHandler._handle_request() done, defined at /home/quinten/.local/lib/python3.10/site-packages/aiohttp/web_protocol.py:423> result=(<Response OK eof>, False) created at /home/quinten/.local/lib/python3.10/site-packages/aiohttp/web_protocol.py:508> took 0.410 seconds
WARNING:asyncio:Executing <Task finished name='AttestationCommunity:on_packet 1' coro=<AttestationCommunity.on_request_attestation() done, defined at /home/quinten/Documents/py-ipv8/ipv8/attestation/wallet/community.py:169> result=None created at /usr/lib/python3.10/asyncio/tasks.py:636> took 0.104 seconds
WARNING:IdentityCommunity:Cannot anonymize community traffic without TunnelEndpoint
WARNING:IdentityCommunity:Cannot anonymize community traffic without TunnelEndpoint
WARNING:asyncio:Executing <Task pending name='Task-250' coro=<IsolatedAsyncioTestCase._asyncioLoopRunner() running at /usr/lib/python3.10/unittest/async_case.py:95> wait_for=<Future pending cb=[Task.task_wakeup()] created at /usr/lib/python3.10/asyncio/base_events.py:429> created at /usr/lib/python3.10/unittest/async_case.py:117> took 0.148 seconds
WARNING:asyncio:Executing <Task finished name='Task-262' coro=<RequestHandler._handle_request() done, defined at /home/quinten/.local/lib/python3.10/site-packages/aiohttp/web_protocol.py:423> result=(<Response OK eof>, False) created at /home/quinten/.local/lib/python3.10/site-packages/aiohttp/web_protocol.py:508> took 0.416 seconds
WARNING:IdentityCommunity:Cannot anonymize community traffic without TunnelEndpoint
WARNING:IdentityCommunity:Cannot anonymize community traffic without TunnelEndpoint
WARNING:asyncio:Executing <Task pending name='Task-300' coro=<IsolatedAsyncioTestCase._asyncioLoopRunner() running at /usr/lib/python3.10/unittest/async_case.py:95> wait_for=<Future pending cb=[Task.task_wakeup()] created at /usr/lib/python3.10/asyncio/base_events.py:429> created at /usr/lib/python3.10/unittest/async_case.py:117> took 0.118 seconds
WARNING:asyncio:Executing <Task finished name='Task-312' coro=<RequestHandler._handle_request() done, defined at /home/quinten/.local/lib/python3.10/site-packages/aiohttp/web_protocol.py:423> result=(<Response OK eof>, False) created at /home/quinten/.local/lib/python3.10/site-packages/aiohttp/web_protocol.py:508> took 0.420 seconds
WARNING:IdentityCommunity:Cannot anonymize community traffic without TunnelEndpoint
WARNING:IdentityCommunity:Cannot anonymize community traffic without TunnelEndpoint
WARNING:asyncio:Executing <Task pending name='Task-328' coro=<IsolatedAsyncioTestCase._asyncioLoopRunner() running at /usr/lib/python3.10/unittest/async_case.py:95> wait_for=<Future pending cb=[Task.task_wakeup()] created at /usr/lib/python3.10/asyncio/base_events.py:429> created at /usr/lib/python3.10/unittest/async_case.py:117> took 0.131 seconds
WARNING:asyncio:Executing <Task finished name='Task-342' coro=<RequestHandler._handle_request() done, defined at /home/quinten/.local/lib/python3.10/site-packages/aiohttp/web_protocol.py:423> result=(<Response OK eof>, False) created at /home/quinten/.local/lib/python3.10/site-packages/aiohttp/web_protocol.py:508> took 0.400 seconds
WARNING:IdentityCommunity:Cannot anonymize community traffic without TunnelEndpoint
WARNING:IdentityCommunity:Cannot anonymize community traffic without TunnelEndpoint
WARNING:asyncio:Executing <Task pending name='Task-350' coro=<IsolatedAsyncioTestCase._asyncioLoopRunner() running at /usr/lib/python3.10/unittest/async_case.py:95> wait_for=<Future pending cb=[Task.task_wakeup()] created at /usr/lib/python3.10/asyncio/base_events.py:429> created at /usr/lib/python3.10/unittest/async_case.py:117> took 0.157 seconds
WARNING:asyncio:Executing <Task finished name='Task-362' coro=<RequestHandler._handle_request() done, defined at /home/quinten/.local/lib/python3.10/site-packages/aiohttp/web_protocol.py:423> result=(<Response OK eof>, False) created at /home/quinten/.local/lib/python3.10/site-packages/aiohttp/web_protocol.py:508> took 0.473 seconds
WARNING:IdentityCommunity:Cannot anonymize community traffic without TunnelEndpoint
WARNING:IdentityCommunity:Cannot anonymize community traffic without TunnelEndpoint
WARNING:asyncio:Executing <Task pending name='Task-414' coro=<IsolatedAsyncioTestCase._asyncioLoopRunner() running at /usr/lib/python3.10/unittest/async_case.py:95> wait_for=<Future pending cb=[Task.task_wakeup()] created at /usr/lib/python3.10/asyncio/base_events.py:429> created at /usr/lib/python3.10/unittest/async_case.py:117> took 0.228 seconds
WARNING:asyncio:Executing <Task finished name='Task-428' coro=<RequestHandler._handle_request() done, defined at /home/quinten/.local/lib/python3.10/site-packages/aiohttp/web_protocol.py:423> result=(<Response OK eof>, False) created at /home/quinten/.local/lib/python3.10/site-packages/aiohttp/web_protocol.py:508> took 0.431 seconds
WARNING:IdentityCommunity:Cannot anonymize community traffic without TunnelEndpoint
WARNING:IdentityCommunity:Cannot anonymize community traffic without TunnelEndpoint
WARNING:asyncio:Executing <Task pending name='Task-442' coro=<IsolatedAsyncioTestCase._asyncioLoopRunner() running at /usr/lib/python3.10/unittest/async_case.py:95> wait_for=<Future pending cb=[Task.task_wakeup()] created at /usr/lib/python3.10/asyncio/base_events.py:429> created at /usr/lib/python3.10/unittest/async_case.py:117> took 0.132 seconds
WARNING:IdentityCommunity:Cannot anonymize community traffic without TunnelEndpoint
WARNING:asyncio:Executing <Task finished name='Task-462' coro=<RequestHandler._handle_request() done, defined at /home/quinten/.local/lib/python3.10/site-packages/aiohttp/web_protocol.py:423> result=(<Response OK eof>, False) created at /home/quinten/.local/lib/python3.10/site-packages/aiohttp/web_protocol.py:508> took 0.413 seconds
WARNING:IdentityCommunity:Cannot anonymize community traffic without TunnelEndpoint
WARNING:IdentityCommunity:Cannot anonymize community traffic without TunnelEndpoint
WARNING:asyncio:Executing <Task pending name='Task-485' coro=<IsolatedAsyncioTestCase._asyncioLoopRunner() running at /usr/lib/python3.10/unittest/async_case.py:95> wait_for=<Future pending cb=[Task.task_wakeup()] created at /usr/lib/python3.10/asyncio/base_events.py:429> created at /usr/lib/python3.10/unittest/async_case.py:117> took 0.132 seconds
WARNING:IdentityCommunity:Cannot anonymize community traffic without TunnelEndpoint
WARNING:IdentityCommunity:Cannot anonymize community traffic without TunnelEndpoint
WARNING:asyncio:Executing <Task pending name='Task-499' coro=<IsolatedAsyncioTestCase._asyncioLoopRunner() running at /usr/lib/python3.10/unittest/async_case.py:95> wait_for=<Future pending cb=[Task.task_wakeup()] created at /usr/lib/python3.10/asyncio/base_events.py:429> created at /usr/lib/python3.10/unittest/async_case.py:117> took 0.146 seconds
WARNING:IdentityCommunity:Cannot anonymize community traffic without TunnelEndpoint
WARNING:IdentityCommunity:Cannot anonymize community traffic without TunnelEndpoint
WARNING:asyncio:Executing <Task pending name='Task-519' coro=<IsolatedAsyncioTestCase._asyncioLoopRunner() running at /usr/lib/python3.10/unittest/async_case.py:95> wait_for=<Future pending cb=[Task.task_wakeup()] created at /usr/lib/python3.10/asyncio/base_events.py:429> created at /usr/lib/python3.10/unittest/async_case.py:117> took 0.149 seconds
WARNING:IdentityCommunity:Cannot anonymize community traffic without TunnelEndpoint
WARNING:IdentityCommunity:Cannot anonymize community traffic without TunnelEndpoint
WARNING:asyncio:Executing <Task pending name='Task-535' coro=<IsolatedAsyncioTestCase._asyncioLoopRunner() running at /usr/lib/python3.10/unittest/async_case.py:95> wait_for=<Future pending cb=[Task.task_wakeup()] created at /usr/lib/python3.10/asyncio/base_events.py:429> created at /usr/lib/python3.10/unittest/async_case.py:117> took 0.154 seconds
WARNING:IdentityCommunity:Cannot anonymize community traffic without TunnelEndpoint
WARNING:IdentityCommunity:Cannot anonymize community traffic without TunnelEndpoint
WARNING:asyncio:Executing <Task pending name='Task-569' coro=<IsolatedAsyncioTestCase._asyncioLoopRunner() running at /usr/lib/python3.10/unittest/async_case.py:95> wait_for=<Future pending cb=[Task.task_wakeup()] created at /usr/lib/python3.10/asyncio/base_events.py:429> created at /usr/lib/python3.10/unittest/async_case.py:117> took 0.212 seconds
WARNING:IdentityCommunity:Cannot anonymize community traffic without TunnelEndpoint
WARNING:IdentityCommunity:Cannot anonymize community traffic without TunnelEndpoint
WARNING:asyncio:Executing <Task pending name='Task-583' coro=<IsolatedAsyncioTestCase._asyncioLoopRunner() running at /usr/lib/python3.10/unittest/async_case.py:95> wait_for=<Future pending cb=[Task.task_wakeup()] created at /usr/lib/python3.10/asyncio/base_events.py:429> created at /usr/lib/python3.10/unittest/async_case.py:117> took 0.149 seconds
WARNING:IdentityCommunity:Cannot anonymize community traffic without TunnelEndpoint
WARNING:IdentityCommunity:Cannot anonymize community traffic without TunnelEndpoint
WARNING:asyncio:Executing <Task pending name='Task-623' coro=<IsolatedAsyncioTestCase._asyncioLoopRunner() running at /usr/lib/python3.10/unittest/async_case.py:95> wait_for=<Future pending cb=[Task.task_wakeup()] created at /usr/lib/python3.10/asyncio/base_events.py:429> created at /usr/lib/python3.10/unittest/async_case.py:117> took 0.165 seconds
WARNING:IdentityCommunity:Cannot anonymize community traffic without TunnelEndpoint
WARNING:IdentityCommunity:Cannot anonymize community traffic without TunnelEndpoint
WARNING:asyncio:Executing <Task pending name='Task-640' coro=<IsolatedAsyncioTestCase._asyncioLoopRunner() running at /usr/lib/python3.10/unittest/async_case.py:95> wait_for=<Future pending cb=[Task.task_wakeup()] created at /usr/lib/python3.10/asyncio/base_events.py:429> created at /usr/lib/python3.10/unittest/async_case.py:117> took 0.183 seconds
WARNING:IdentityCommunity:Cannot anonymize community traffic without TunnelEndpoint
WARNING:IdentityCommunity:Cannot anonymize community traffic without TunnelEndpoint
WARNING:asyncio:Executing <Task pending name='Task-657' coro=<IsolatedAsyncioTestCase._asyncioLoopRunner() running at /usr/lib/python3.10/unittest/async_case.py:95> wait_for=<Future pending cb=[Task.task_wakeup()] created at /usr/lib/python3.10/asyncio/base_events.py:429> created at /usr/lib/python3.10/unittest/async_case.py:117> took 0.184 seconds
WARNING:IdentityCommunity:Cannot anonymize community traffic without TunnelEndpoint
WARNING:IdentityCommunity:Cannot anonymize community traffic without TunnelEndpoint
WARNING:asyncio:Executing <Task pending name='Task-673' coro=<IsolatedAsyncioTestCase._asyncioLoopRunner() running at /usr/lib/python3.10/unittest/async_case.py:95> wait_for=<Future pending cb=[Task.task_wakeup()] created at /usr/lib/python3.10/asyncio/base_events.py:429> created at /usr/lib/python3.10/unittest/async_case.py:117> took 0.163 seconds
WARNING:asyncio:Executing <Task finished name='Task-687' coro=<RequestHandler._handle_request() done, defined at /home/quinten/.local/lib/python3.10/site-packages/aiohttp/web_protocol.py:423> result=(<Response OK eof>, False) created at /home/quinten/.local/lib/python3.10/site-packages/aiohttp/web_protocol.py:508> took 0.403 seconds
WARNING:IdentityCommunity:Cannot anonymize community traffic without TunnelEndpoint
WARNING:IdentityCommunity:Cannot anonymize community traffic without TunnelEndpoint
WARNING:asyncio:Executing <Task pending name='Task-695' coro=<IsolatedAsyncioTestCase._asyncioLoopRunner() running at /usr/lib/python3.10/unittest/async_case.py:95> wait_for=<Future pending cb=[Task.task_wakeup()] created at /usr/lib/python3.10/asyncio/base_events.py:429> created at /usr/lib/python3.10/unittest/async_case.py:117> took 0.142 seconds
WARNING:asyncio:Executing <Task finished name='Task-709' coro=<RequestHandler._handle_request() done, defined at /home/quinten/.local/lib/python3.10/site-packages/aiohttp/web_protocol.py:423> result=(<Response OK eof>, False) created at /home/quinten/.local/lib/python3.10/site-packages/aiohttp/web_protocol.py:508> took 0.449 seconds
WARNING:IdentityCommunity:Cannot anonymize community traffic without TunnelEndpoint
WARNING:IdentityCommunity:Cannot anonymize community traffic without TunnelEndpoint
WARNING:asyncio:Executing <Task pending name='Task-717' coro=<IsolatedAsyncioTestCase._asyncioLoopRunner() running at /usr/lib/python3.10/unittest/async_case.py:95> wait_for=<Future pending cb=[Task.task_wakeup()] created at /usr/lib/python3.10/asyncio/base_events.py:429> created at /usr/lib/python3.10/unittest/async_case.py:117> took 0.146 seconds
WARNING:IdentityCommunity:Cannot anonymize community traffic without TunnelEndpoint
WARNING:asyncio:Executing <Task pending name='Task-717' coro=<IsolatedAsyncioTestCase._asyncioLoopRunner() running at /usr/lib/python3.10/unittest/async_case.py:101> wait_for=<Future pending cb=[Task.task_wakeup()] created at /usr/lib/python3.10/asyncio/base_events.py:429> created at /usr/lib/python3.10/unittest/async_case.py:117> took 0.149 seconds
WARNING:asyncio:Executing <Task finished name='Task-737' coro=<RequestHandler._handle_request() done, defined at /home/quinten/.local/lib/python3.10/site-packages/aiohttp/web_protocol.py:423> result=(<Response OK eof>, False) created at /home/quinten/.local/lib/python3.10/site-packages/aiohttp/web_protocol.py:508> took 0.420 seconds
ERROR:DHTCommunity:Task resulted in error: Value was not stored
Traceback (most recent call last):
  File "/home/quinten/Documents/py-ipv8/ipv8/taskmanager.py", line 131, in done_cb
    future.result()
  File "/home/quinten/Documents/py-ipv8/ipv8/dht/community.py", line 348, in store_on_nodes
    raise DHTError('Value was not stored')
ipv8.dht.DHTError: Value was not stored

ERROR:DHTCommunity:Task resulted in error: Value was not stored
Traceback (most recent call last):
  File "/home/quinten/Documents/py-ipv8/ipv8/taskmanager.py", line 131, in done_cb
    future.result()
  File "/home/quinten/Documents/py-ipv8/ipv8/dht/community.py", line 348, in store_on_nodes
    raise DHTError('Value was not stored')
ipv8.dht.DHTError: Value was not stored

WARNING:RequestCache:Dropping <Request find-12965> due to shutdown!
                                                  done!

Summary:
[SUCCESS] Ran 608 tests in 28.86 seconds (66.57 seconds total in tests).
```

</details>

Nothing very interesting in here and I'm still not advocating to  bring it back. However, this is data _could_ be useful.

Reply:
That's strange. I'm not getting output with or without the patch. I guess it's a Windows vs Linux thing then, since it's running a different `EventLoop`. Anyway, I'll mark this conversation as resolved.